specification.md

```diff
@@ -381,6 +396,112 @@ a resource containing an array of [retry](#Retry-Definition) definition.
 Referenced resource can be used by multiple workflow definitions. For more information about 
 using and referencing retry definitions see the [Workflow Error Handling](#Workflow-Error-Handling) section.
 
+#### ExecWindow Definition
+
+| Parameter | Description | Type | Required |
+| --- | --- | --- | --- |
+| time | Amount of time (ISO 8601 format) in the execution window | string | yes |
+| interrupt | If `false`, workflow instance is allowed to finish current execution. If `true`, current workflow execution is abrupted. Default is `false`  | boolean | no |
+| workflowId | Unique Id of a workflow to be executed before workflow instance is terminated | string | no |
+
+
+<details><summary><strong>Click to view example definition</strong></summary>
+<p>
+
+<table>
+<tr>
+    <th>JSON</th>
+    <th>YAML</th>
+</tr>
+<tr>
+<td valign="top">
+
+```json
+{  
+   "time": "PT2M",
+   "workflowId": "createandsendreport"
+}
+```
+
+</td>
+<td valign="top">
+
+```yaml
+time: PT2M
+workflowId: createandsendreport
+```
+
+</td>
+</tr>
+</table>
+
+</details>
+
+The `time` property defines the total amount of time in the execution window. Once a workflow instance is created,
+and the amount of the defined time is reached, the workflow instance needs to be terminated.
+
+The `interrupt` property defines if the currently running instance should be allowed to finish its current 
+execution flow before it needs to be terminated. If set to `true`, the current instance execution must stop immediately.
+
+The `workflowId` property defines an unique id of a sub-workflow which should be executed before the workflow instance
+is terminated.
+
+`ExecWindow` allows us to solve interesting use cases, especially within the IoT (Internet of Things) domain.
```

Comment:
@jorgenj i am working on additional use case involving batch processing where this can be used to cancel a "job" after a certain amount of time. hope this will cover the concern as well

Reply:
added two new examples, added a very similar one as you requested but instead of dealing with patient registration multi-step process its an order confirmation multi-step process. hope this is ok